Sources/SwiftProtobuf/DelimitedMessageUtil.swift

```diff
@@ -0,0 +1,215 @@
+// Sources/SwiftProtobuf/DelimitedMessageUtil.swift - Delimited support
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/master/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+///
+/// Helpers to read/write message with a length prefix.
+///
+// -----------------------------------------------------------------------------
+
+import Foundation
+
+/// Additional errors for delimited message handing.
+public enum DelimitedMessageError: Error {
+  /// If a read/write to the stream fails, but the stream's `streamError` is nil,
+  /// this error will be throw instead since the stream didn't provide anything
+  /// more specific. A common cause for this can be failing to open the stream
+  /// before trying to read/write to it.
+  case unknownStreamError
+
+  /// While reading/writing to the stream, less than the expected bytes was
+  /// read/written.
+  case truncated
+}
+
+/// Helper methods for reading/writing messages with a length prefix.
+public enum DelimitedMessageUtil {
```

Comment:
Or perhaps `DelimitedMessageStream`?  In the sense that this is something you write messages to or read them from.

Another structure would be to have `DelimitedMessageOutputStream` and `DelimitedMessageInputStream` that you used like so:
```
  let binaryStream: OutputStream
  let messageStream = DelimitedMessageOutputStream(stream: binaryStream)
  messageStream.write(message)
```